### PR TITLE
workflows: Update build workflow to not rely on curl based upload

### PR DESCRIPTION
**Pull request by hjpotter92**
- getting our build uploads flow to be same as other projects under `livepeer/` organisation
- replacing the `upload_build.sh` script with python equivalent to only handle discord notification, uploads would be done as github action
- combine discord notification to a single message per commit instead of separate message per individual build platform/architecture combo

screenshot of difference between successful build notifications on discord (channel: [`#builds`](https://discord.com/channels/423160867534929930/577736983036559360)):

### before:

image

### after:

image

### CODE REVIEW

File: .github/discord-embed-webhook.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+def get_embeds(embed: DiscordEmbed, ref_name: str, checksums: list[str]):
+    for line in checksums:
+        shasum, filename = line.split()
+        download_url = DOWNLOAD_BASE_URL_TEMPLATE.format(
+            version=ref_name,
+            filename=filename,
+        )
+        title = filename[9:].split(".")[0]
```
Comment (pwilczynskiclearcode) [inline .github/discord-embed-webhook.py:22]: A comment with example why we take 9th on letter would be helpful

File: .github/discord-embed-webhook.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        color=2928914,
+        url=commit_url,
+    )
+    embed.add_embed_field(name="Commit SHA", value=head.get("sha"))
+    embed.set_author(name=args.git_committer)
+    get_embeds(embed, args.ref_name, checksums)
+    embed.set_timestamp()
+    webhook.add_embed(embed)
+    response = webhook.execute()
+    print(response)
```
Comment (pwilczynskiclearcode) [inline .github/discord-embed-webhook.py]: Should we check the response status and assert if it's 2XX? just printing errors won't fail the gihtub workflow

**Comment by pwilczynskiclearcode** [comment]
A ticket or discord discussion link would help to understand what's the problem that you're solving

**Comment by hjpotter92** [comment]
> A ticket or discord discussion link would help to understand what's the problem that you're solving

it's present in the PR's message above :)

File: .github/discord-embed-webhook.py
Comment (pwilczynskiclearcode) [inline .github/discord-embed-webhook.py:1]: This is sth new? Unrelated to the upload issue?

**Comment by pwilczynskiclearcode** [comment]
> > A ticket or discord discussion link would help to understand what's the problem that you're solving
> 
> it's present in the PR's message above :)

I know. I just added it ;)

File: .github/discord-embed-webhook.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+def get_embeds(embed: DiscordEmbed, ref_name: str, checksums: list[str]):
+    for line in checksums:
+        _, filename = line.split()
+        download_url = DOWNLOAD_BASE_URL_TEMPLATE.format(
+            version=ref_name,
+            filename=filename,
+        )
+        # strip the initial `livepeer-` prefix and then identify build
+        # platform/architecture/release-tag values from file name
+        title = filename[9:].split(".")[0]
```
Comment (pwilczynskiclearcode) [inline .github/discord-embed-webhook.py]: I see. I think it could be written using [lstrip()](https://docs.python.org/3/library/stdtypes.html#str.lstrip)
```python
# strip the initial `livepeer-` prefix and then identify build
# platform/architecture/release-tag values from file name
title = filename.lstrip("livepeer-").split(".")
```

File: .github/discord-embed-webhook.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        url=commit_url,
+    )
+    embed.add_embed_field(name="Commit SHA", value=head.get("sha"))
+    embed.set_author(name=args.git_committer)
+    get_embeds(embed, args.ref_name, checksums)
+    embed.set_timestamp()
+    webhook.add_embed(embed)
+    response: Response = webhook.execute()
+    # Fail the script if discord returns anything except OK status
+    assert response.ok, "Discord webhook failed"
```
Comment (pwilczynskiclearcode) [inline .github/discord-embed-webhook.py]:
```python
assert response.ok, f"Discord webhook failed {response.status} {response.text}"
```

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -289,16 +290,33 @@ jobs:
...
         id: upload-manifest
-        uses: google-github-actions/upload-cloud-storage@v1
+        uses: google-github-actions/upload-cloud-storage@v2
         with:
           path: ${{ steps.branch-manifest.outputs.manifest-file }}
           destination: "build.livepeer.live/${{ github.event.repository.name }}/"
           parent: false
+          process_gcloudignore: false
+
+      - name: Trigger discord webhook
```
Comment (pwilczynskiclearcode) [inline .github/workflows/build.yaml:309]: Which Discord channel receives those messages? A comment in code would be nice

**Review by pwilczynskiclearcode — APPROVED** [review]

**Comment by j0sh** [comment]
> it's present in the PR's message above :)

It would be better to have the description directly in the ticket, there is a template for that. Discord is not easily accessible and the conversation is behind a private channel, but this is an open source repo.

File: .github/workflows/build.yaml [hunk 6]
```diff
@@ -120,7 +111,6 @@ jobs:
         run: go mod download
 
       - name: Install ffmpeg
-        if: steps.cache-ffmpeg.outputs.cache-hit != 'true'
```
Comment (j0sh) [inline .github/workflows/build.yaml:123]: curious why this is being removed since ffmpeg takes a long time to build

also seems unrelated to the rest of the PR

File: .github/discord-embed-webhook.py
Comment (hjpotter92) [reply .github/discord-embed-webhook.py:1]: the upload issue is because gcp deprecated its signature mechanism we were relying on in `upload_build.sh` script. it was also sending out separate build messages for each build to the discord channel instead of grouping them together.

File: .github/discord-embed-webhook.py (hunk 3 above)
Comment (hjpotter92) [reply .github/discord-embed-webhook.py]: done*

File: .github/discord-embed-webhook.py (hunk 4 above)
Comment (hjpotter92) [reply .github/discord-embed-webhook.py]: done*

File: .github/workflows/build.yaml (hunk 6 above)
Comment (hjpotter92) [reply .github/workflows/build.yaml:123]: the cache has not been working forever (screenshot below)

image

i tried fixing it in [`3f048d0` (#3117)](https://github.com/livepeer/go-livepeer/pull/3117/commits/3f048d052147045bb684cc1258af431fb01e364d) but that led to the [next builds failing to find `protoc`](https://github.com/livepeer/go-livepeer/actions/runs/10282216718/job/28453448609?pr=3117). no idea why skipping ffmpeg install step leads to failure to find protoc (and we've observed this happen in the past too). would appreciate some insight into fixing the ffmpeg installation script if we can make use of caching there.

**Comment by hjpotter92** [comment]
> > it's present in the PR's message above :)
> 
> It would be better to have the description directly in the ticket, there is a template for that. Discord is not easily accessible and the conversation is behind a private channel, but this is an open source repo.

updated the PR description. it doesnt follow the template structure, but did list out most of the changes covered.

File: .github/workflows/build.yaml (hunk 6 above)
Comment (j0sh) [reply .github/workflows/build.yaml:123]: Thanks for the context - I am not too familiar with the CI build actions myself. The only thing I can think of is there is a [v4 update](https://github.com/actions/cache?tab=readme-ov-file#v4) to the action

> skipping ffmpeg install step leads to failure to find protoc

weird, maybe it implicitly installs protoc somehow ... but I am not seeing anything obvious in the install script (and my own box doesn't even have it)

**Review by leszko — APPROVED** [review]
